From 29346eec86139f79ba2157c01da666753426b4c2 Mon Sep 17 00:00:00 2001
From: Achill Gilgenast <achill@achill.org>
Date: Mon, 10 Nov 2025 16:38:45 +0100
Subject: [PATCH] Revert "Remove support for elogind"

(e)logind is used by fwupd for poweroff/reboot and for the logind
plugin, which "is used to ensure that the machine does not enter a
low power mode when updates are being performed".

Upstream removed support for elogind, but I think it is still useful and
important downstream.

This reverts commit bbda9dec06674757da55ce6a241953c37553aaf9.

Signed-off-by: Achill Gilgenast <achill@achill.org>
---
 meson.build                | 6 ++++++
 meson_options.txt          | 5 +++++
 plugins/logind/meson.build | 2 +-
 3 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index b711ab04b031..ed039263beb6 100644
--- a/meson.build
+++ b/meson.build
@@ -713,6 +713,11 @@ if libsystemd.found()
   endif
 endif
 
+elogind = dependency('systemd', 'libelogind', required: get_option('elogind'))
+if elogind.found()
+  conf.set('HAVE_LOGIND' , '1')
+endif
+
 supported_build = get_option('supported_build').disable_auto_if(not tag).allowed()
 if supported_build
   conf.set('SUPPORTED_BUILD', '1')
@@ -939,6 +944,7 @@ summary(
     'dbus_socket_address': get_option('dbus_socket_address'),
     'vendor_ids_dir': vendor_ids_dir,
     'docs': build_docs,
+    'elogind': elogind,
     'gnutls': gnutls,
     'introspection': introspection.allowed(),
     'libblkid': libblkid,
diff --git a/meson_options.txt b/meson_options.txt
index 18991b9837dd..1facbc39e4ed 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -48,6 +48,11 @@ option(
   type: 'string',
   description: 'the hardcoded name of OS directory in ESP, e.g. fedora',
 )
+option(
+  'elogind',
+  type: 'feature',
+  description: 'elogind support',
+)
 option(
   'firmware-packager',
   type: 'boolean',
diff --git a/plugins/logind/meson.build b/plugins/logind/meson.build
index bd66fabbc228..7fe59f5945cc 100644
--- a/plugins/logind/meson.build
+++ b/plugins/logind/meson.build
@@ -1,4 +1,4 @@
-libsystemd.found() or subdir_done()
+libsystemd.found() or elogind.found() or subdir_done()
 
 cargs = ['-DG_LOG_DOMAIN="FuPluginLogind"']
 plugins += {meson.current_source_dir().split('/')[-1]: true}
-- 
2.51.2

